spec/components/auth/github_authenticator_spec.rb

```diff
@@ -169,4 +182,30 @@
     end
 
   end
+
+  describe 'avatar retrieval' do
+    context 'when user has a custom avatar' do
+      let(:user_avatar) { Fabricate(:user_avatar, custom_upload: Fabricate(:upload)) }
+      let(:user_with_custom_avatar) { Fabricate(:user, user_avatar: user_avatar) }
+
+      it 'does not enqueue a download_avatar_from_url job' do
+        Jobs.expects(:enqueue).with(:download_avatar_from_url, anything).never
```

Comment:
Instead of mocking, lets configure `SiteSetting.queue_jobs = false` and test for the actual job in the queue. This may be helpful https://github.com/mperham/sidekiq/wiki/Testing